mondey_backend/src/mondey_backend/routers/admin_routers/milestones.py

```diff
@@ -69,11 +71,53 @@ def update_milestone_group_admin(
         return db_milestone_group
 
     @router.delete("/milestone-groups/{milestone_group_id}")
-    def delete_milestone_group_admin(session: SessionDep, milestone_group_id: int):
+    def delete_milestone_group_admin(
+        session: SessionDep,
+        milestone_group_id: int,
+        dry_run: bool = Query(
+            True,
+            description="When true, shows what would be deleted without actually deleting",
+        ),
+    ):
         milestone_group = get(session, MilestoneGroup, milestone_group_id)
-        session.delete(milestone_group)
-        session.commit()
-        return {"ok": True}
+
+        if not dry_run:
+            affected_milestone_answers = 0
+            groups_milestone_ids = [
+                milestone.id
+                for milestone in milestone_group.milestones
+                if milestone.id is not None
+            ]
+
+            for milestone_id in groups_milestone_ids:
+                affected_milestone_answers += count_milestone_answers_for_milestone(
+                    session, milestone_id
+                )
+
+            session.delete(milestone_group)
+            session.commit()
+            return {
+                "ok": True,
+                "deletion_executed": True,
+                "deleted_milestone_count": len(groups_milestone_ids),
+                "deleted_answer_count": affected_milestone_answers,
+            }
+
+        affectedAnswers = 0
+        for milestone in milestone_group.milestones:
+            if milestone.id is None:
+                continue
+            affectedAnswers += count_milestone_answers_for_milestone(
+                session, milestone.id
+            )
```

Comment:
here the logic to count affectedAnswers is duplicated in both branches - could this be done once before the `if dry_run` branch?